service/authenticator.go

```diff
@@ -18,7 +17,11 @@ import (
 // Email Token is also authenticated to complete email verification.
 // Return a context used for the authentication.
 func Authenticate(ctx context.Context) (newCtx context.Context, err error) {
-	log.Info(consts.AuthTag, fmt.Sprintf("%v", ctx))
+	log.RequestService(consts.AuthTag)
+	if ctx == nil {
+		log.Error(consts.AuthTag, consts.ErrNilContext.Error())
+		return context.TODO(), consts.ErrNilContext
```

Comment:
what does context.TODO() means ?

Reply:
@dinhl 
`context.TODO()` function returns an empty context as well. TODO is used while refactoring functions to support context. We use it when a parent context is not available in that function yet

https://golang.org/pkg/context/#TODO

Most likely it will be replaced in the future.